#!/usr/bin/env python
PACKAGE = "rm_shooter_controllers"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add("block_effort", double_t, 0, "Trigger block effort", 0.95, 0.0, 10)
gen.add("block_duration", double_t, 0, "Trigger block duration", 0.05, 0.0, 2.0)
gen.add("block_speed", double_t, 0, "Trigger block speed", 0.5, 0.0, 5)
gen.add("anti_block_angle", double_t, 0, "Trigger anti block angle", 0.35, 0.0, 0.785)
gen.add("push_angle", double_t, 0, "Trigger push angle", 0.785, 0.0, 1.5)
gen.add("anti_block_error", double_t, 0, "Trigger anti block error", 0.05, 0.0, 0.2)
gen.add("magazine_q_des", double_t, 0, "Magazine q_des", 0.5, 0.0, 5.0)
gen.add("qd_10", double_t, 0, "Friction wheel rotation speed when bullet speed is 10m/s", 300, 0.0, 9999)
gen.add("qd_15", double_t, 0, "Friction wheel rotation speed when bullet speed is 15m/s", 300, 0.0, 9999)
gen.add("qd_16", double_t, 0, "Friction wheel rotation speed when bullet speed is 16m/s", 300, 0.0, 9999)
gen.add("qd_18", double_t, 0, "Friction wheel rotation speed when bullet speed is 18m/s", 300, 0.0, 9999)
gen.add("qd_30", double_t, 0, "Friction wheel rotation speed when bullet speed is 30m/s", 300, 0.0, 9999)

exit(gen.generate(PACKAGE, "shooter", "ShooterBase"))
